src/js/components/Legenda/index.js

```diff
@@ -33,9 +37,53 @@ const legendas = [
     },
 ];
 
-export function encontraLegenda(label) {
-
+function encontraLegenda(label) {
     const legendaEncontrada = legendas.find(legenda => legenda.label === label);
 
     return legendaEncontrada && legendaEncontrada.legenda;
-}
\ No newline at end of file
+}
+
+export default class Legenda extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
```

Comment:
we don't need state in this component, I guess just the prop `labels` it's enough.